src/content/en/2021/caching.md

```diff
@@ -17,4 +20,425 @@ featured_stat_3: TODO
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1359015817&format=interactive",
+  sheets_gid="1944529311",
+  sql_file="cache_control_directives.sql"
+) }}
+
+[TODO Leo complete the part around Cache-Control directives]
+Cache Control directives, 61% of mobile requests include a Cache-Control response header with a `max-age` directive.
+
+The most misused Cache Control directive is `set-cookie`, used for 0.07% of total directives for Desktop and 0.08% for Mobile.
+
+Fun fact the the largest max-age value is 1625540701883604800000 seconds or 51545557517871.8 years (51 trillion).
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="big-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+[TODO Leo talk about `immutable`]
+Talk about `immutable` directive, 12% of desktop requests containing Cache-Control: immutable are on the "static.parastorage.com" host.
+
+### `Last-Modified` & `ETag` adoption
+
+[TODO Leo]
+
+### `304` Not Modified status
+
+304 responses are usually much smaller than 200 responses, so we can see an improvement in page performance when we use conditional requests correctly. Revalidation on conditional requests can only be done by using either an `Etag` or `Last-Modified` header. 
+
+The distribution of 304 responses increased by 7.7% for `If-Modified-Since` in 2021, compared to 2020.
+
+{{ figure_markup(
+  image="http-304-by-caching-strategy.png",
+  caption="HTTP 304 response rate by caching strategy.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1965033294&format=interactive",
+  sheets_gid="1136058277",
+  sql_file="valid_if_none_match_returns_304.sql"
+) }}
+
+[TODO Leo: Better explore this part!!!]
+
+### Validity of date strings
+
+The three main HTTP headers used to represent timestamps, `Date`,`Last-Modified` and `Expires` all use a date formatted string. The `Date` HTTP response header is almost always generated automatically by the web server, meaning that invalid values are extremely rare. Still, in the rare event that the date is set incorrectly it can affect cacheability on the response on which it is served. 
+
+[TODO Add figure showing on Mobile Invalid date strings between pct_using_invalid_date, pct_using_invalid_last_modified and  pct_using_invalid_expires, data from 2020 https://docs.google.com/spreadsheets/d/1fYmpSN3diOiFrscS75NsjfsrKXzxxhUMNcYSqXnQJQU/edit#gid=1338572274 and 2021 https://docs.google.com/spreadsheets/d/1-v3yR0LZIC3t4zWtqTgR3jJsKjjRMP-HATU2caP8e2c/edit#gid=1680471251 Potentiall we mighe want to show 2020 vs 2021]
+
+Between 2020 and 2021, the percent using invalid date improved by a .5% but got worse for last-modified and expires showing that it was related to how the date was set on caching.
```

Comment:
Edit: 

Between 2020 and 2021, the percent using invalid `Date` improved by 0.5% but worsened for `Last-Modified` and `Expires` showing that it was related to how the date was set on caching. 

This shows us that automation of the `Date` header could benefit from further attention...?